generated/nidaqmx/nidaqmx_client.h

```diff
@@ -0,0 +1,7767 @@
+
+//---------------------------------------------------------------------
+// This file is automatically generated. All manual edits will be lost.
+//---------------------------------------------------------------------
+// Client convenience header for the NI-DAQMX Metadata
+//---------------------------------------------------------------------
+#ifndef NIDAQMX_GRPC_CLIENT_H
+#define NIDAQMX_GRPC_CLIENT_H
+
+#include <grpcpp/grpcpp.h>
+
+#include <nidaqmx.grpc.pb.h>
+
+#include <cstdint>
+#include <stdexcept>
+#include <vector>
+
+namespace nidaqmx_grpc::experimental::client {
+
+using namespace nidaqmx_grpc;
+
+template <typename TFirst, typename TSecond>
+class simple_variant
```

Comment:
This is a trick to make the oneofs for raw enums work. Since this type supports implicit conversion from either TFirst or TSecond, methods taking a simple_variant will accept either argument and do the right thing. That's the theory anyway ...

This is the same behavior std::variant would have but std::variant doesn't work with out linux rt compiler.

The easiest alternative would be to only support the non-raw params, although that's limiting for the attribute accessors that often use both (and where we have more work planned for raw attribute IDs).

Pass both by pointer would work, but it's annoying for callsites. I think passing both as std::optional would also work, but we'd probably need a simple_optional for the same reason as simple_variant.